proto/buf/alpha/registry/v1alpha1/image.proto

```diff
@@ -34,3 +37,53 @@ message GetImageRequest {
 message GetImageResponse {
   buf.alpha.image.v1.Image image = 1;
 }
+
+enum PartialImageLevel {
+  PARTIAL_IMAGE_LEVEL_UNSPECIFIED = 0;
+
+  // Returns complete files in which the requested types are contained.
+  PARTIAL_IMAGE_LEVEL_FILES = 1;
+
+  // Returns trimmed files in which the requested types are contained. In a
+  // trimmed file, all declarations that are not embedded (directly or
+  // indirectly) in requested types are removed.
+  PARTIAL_IMAGE_LEVEL_TYPES = 2;
+}
```

Comment:
As far as I understand, this is a remnant of an earlier version? I think the most robust and expected behavior would be to keep types in their original files.

Reply:
I forgot to include this as a field, it is something that Alex and I discussed to get more minimal filedescriptors that only contain the types you're interested in without returning types just because they were declared in the same file.

I agree with you that this is much harder to get right. Especially rewriting `SourceCodeInfo` and removeing unused imports as if those types were never there will be tricky. Additionally, merging multiple of these received over multiple client calls on the client side is painful, whereas if you only have to combine complete filedescriptors its much more straightforward.

I don't think it needs to be included and agreed on today. Perhaps if we want this very granular type information we should look for other avenues than the Images/FileDescriptor to communicate that.